File: code-examples/verifyKyaPayToken/python/validators/kya.py
import validators


def validate_kya_claims(payload):
    """
    Validate KYA identity claims.
    This example shows basic validation (email); enforce additional checks as needed.
    """
    # hid.email is a valid email address
    email = payload.get("hid", {}).get("email")
    if not validators.email(email):
        print("Invalid email format")
        return {"success": False, "error": "invalid_email", "message": "Invalid email format."}
    return None

File: code-examples/verifyKyaPayToken/typescript/src/validators/kya.ts
import type { JWTPayload } from "jose";
import validator from "validator";
import type { VerifyError } from "../utils.js";

/**
 * Validate KYA identity claims.
 * This example shows basic validation (email); enforce additional checks as needed.
 */
export function validateKyaClaims(payload: JWTPayload): VerifyError | null {
  const email = (payload as any)?.hid?.email;
  // email is a valid email address
  if (!validator.isEmail(String(email))) {
    console.error("Invalid email format");
    return { success: false, error: "invalid_email", message: "Invalid email format." };
  }
  return null;
}
